diagnostics/model_test/verify_model.py

```diff
@@ -0,0 +1,161 @@
+import json
+import os
+
+
+def calculate_output(param_dict, select_device, input_example):
+    """Calculate the output of the imported frozen graph given the input.
+
+    Load the graph def from frozen_graph_file on selected device, then get the tensors based on the input and output name from the graph,
+    then feed the input_example to the graph and retrieves the output vector.
+
+    Args:
+    param_dict: The dictionary contains all the user-input data in the json file.
+    select_device: "NGRAPH" or "CPU".
+    input_example: Random generated input or actual image.
+
+    Returns:
+        The output vector obtained from running the input_example through the graph.
+    """
+    frozen_graph_filename = param_dict["frozen_graph_location"]
+    input_tensor_name = param_dict["input_tensor_name"]
+    output_tensor_name = param_dict["output_tensor_name"]
+
+    with tf.gfile.GFile(frozen_graph_filename, "rb") as f:
+        graph_def = tf.GraphDef()
+        graph_def.ParseFromString(f.read())
+
+    with tf.Graph().as_default() as graph:
+        with tf.device('/job:localhost/replica:0/task:0/device:' + select_device + ':0'):
+            tf.import_graph_def(graph_def)
+
+    input_placeholder = graph.get_tensor_by_name(input_tensor_name)
+    output_tensor = graph.get_tensor_by_name(output_tensor_name)
+
+    config = tf.ConfigProto(
+        allow_soft_placement=True,
+        # log_device_placement=True,
+        inter_op_parallelism_threads=1
+    )
+
+    with tf.Session(graph=graph, config=config) as sess:
+        output_tensor = sess.run(output_tensor, feed_dict={
+                                 input_placeholder: input_example})
+        return output_tensor
+
+
+def calculate_norm(ngraph_output, tf_output, desired_norm):
+    """Calculate desired_norm between vectors.
+
+    Calculate the L1/L2/inf norm between the NGRAPH and tensorflow output vectors.
+
+    Args:
+        ngraph_output: The output vector generated from NGRAPH graph.
+        tf_output: The output vector generated from tensorflow graph.
+        desired_norm: L1/L2/inf norm. 
+
+    Returns:
+        Calcualted norm between the vectors.
+
+    Raises:
+        Exception: If the dimension of the two vectors mismatch.
+    """
+    if(ngraph_output.shape != tf_output.shape):
+        raise Exception('ngraph output and tf output dimension mismatch')
+
+    ngraph_output_squeezed = np.squeeze(ngraph_output)
+    tf_output_squeezed = np.squeeze(tf_output)
+
+    ngraph_output_flatten = ngraph_output_squeezed.flatten()
+    tf_output_flatten = tf_output_squeezed.flatten()
+
+    factor = np.prod(ngraph_output_squeezed.shape)
+
+    if desired_norm == 'l1_norm':
+        return np.sum(np.abs(ngraph_output_flatten - tf_output_flatten), axis=0) / factor
+    elif desired_norm == 'l2_norm':
+        return np.sum(np.dot(np.abs(ngraph_output_flatten - tf_output_flatten), np.abs(ngraph_output_flatten - tf_output_flatten)))/factor
+    elif desired_norm == 'inf_norm':
```

Comment:
you could just use np.linalg.norm for the other norms too.

in that case desired_norm can be np.inf, or 1 or 2. In which case we just call it like below for all cases:
```
if desired_norm not in [1,2,np.inf]:
    Exception("Only 1, 2 and inf norms supported")
np.linalg.norm((ngraph_output_flatten - tf_output_flatten), desired_norm)```